cli/src/main/java/com/devonfw/tools/ide/commandlet/ShellCommandlet.java

```diff
@@ -86,6 +90,10 @@ public void run() {
         while (true) {
           try {
             line = reader.readLine(prompt, rightPrompt, (MaskingCallback) null, null);
+            line = line.trim();
```

Comment:
Is this required for this feature?

Reply:
The tab of the auto completion is registered as a character in the line-string so if we want to match the string with "exit" we need to either trim it down or use another method like `line.contains("exit")` however these would probably also match some nonsense like : "fireexit" . If there is another method we could easily use here i am happy to now about.

Reply:
You could do `if (line.trim().equals("exit"))` if the `trim` is undesired for the rest of the code.
However, I do not see this `trim()` as bad but rather as a general improvement since also `runCommand` should not cause problems in case of a training tab or space.